libbpf-rs/src/map.rs

```diff
@@ -455,10 +455,18 @@ impl Map {
 
         let mut out: Vec<u8> = Vec::with_capacity(out_size);
 
+        let mut map_key = key.as_ptr() as *const c_void;
+        if self.key_size() == 0
+            && (matches!(self.map_type(), MapType::Queue)
+                || matches!(self.map_type(), MapType::Stack))
+        {
```

Comment:
```suggestion
if self.key_size() == 0 && matches!(self.map_type(), MapType::Queue | MapType::Stack) {
```

Reply:
^ here and elsewhere

Reply:
In general, would it be possible to introduce a function `map_key` that contains the newly added logic for map key inference?

Reply:
hello @danielocfb, appreciate for the prompt review ! Yes I should have done that to avoid the repetition, it's fixed in the latest push.